tests/test_nearest_neighbors.py

```diff
@@ -114,8 +114,8 @@ def test_example_with_id(gpu_number: int) -> None:
         gpu_knn = gpu_knn.setIdCol("id")
         gpu_knn = gpu_knn.setK(topk)
 
-        gpu_knn = gpu_knn.fit(data_df)
-        query_df, item_df, knn_df = gpu_knn.kneighbors(query_df)
+        gpu_model = gpu_knn.fit(data_df)
```

Comment:
should add a test that asserts that an exception is raised if `transform` is invoked.

Reply:
Added the following:

```python
    def _transform(self, dataset: DataFrame) -> DataFrame:
        raise NotImplementedError(
            "'NearestNeighborsModel does not provide a transform function. Use 'kneighbors' instead."
        )
```

Tested in pyspark shell that the error will be raised if transform is invoked. 

Did not find the solution on how to add assert statement for exception. Let me know.

Reply:
See [this example](https://github.com/NVIDIA/spark-rapids-ml/blob/spark-cuml/tests/test_kmeans.py#L317-L318).

Reply:
Done.